src/test/java/com.teragrep.jos_01.procfs/ProcessTest.java

```diff
@@ -0,0 +1,355 @@
+        Assertions.assertDoesNotThrow(() -> {
+            systemd = new Process(1);
+            Stat stat = systemd.stat();
+            Assertions.assertEquals(1, stat.pid());
+            Assertions.assertEquals("(systemd)", stat.comm());
+            Iterator iterator = stat.read().iterator();
+            int i = 0;
+            while (iterator.hasNext()) {
+                Assertions.assertNotNull(iterator.next());
+                i++;
+            }
+            Assertions.assertEquals(52, i);
+
+        });
+        systemd = new Process(1);
+    }
+
+    // Kthreadd process (pid = 2) should have 52 fields in its Stat file, and it should have (kthreadd) as comm value and 2 as pid value in the file. None of the fields should be null
+    @Test
+    public void readKthreaddFileTest() {
+        Assertions.assertDoesNotThrow(() -> {
+            kthreadd = new Process(2);
+            Stat stat = kthreadd.stat();
+            Assertions.assertEquals(2, stat.pid());
+            Assertions.assertEquals("(kthreadd)", stat.comm());
+            Iterator iterator = stat.read().iterator();
+            int i = 0;
+            while (iterator.hasNext()) {
+                Assertions.assertNotNull(iterator.next());
+                i++;
+            }
+            Assertions.assertEquals(52, i);
+
+        });
+    }
+
+    // Process object should be able to be instantiated with an integer and a String
+    @Test
+    public void constructorTest() {
+        Process process = new Process(1);
+        Process processString = new Process("1");
+        Assertions.assertTrue(process.isAlive());
+        Assertions.assertTrue(processString.isAlive());
+    }
+
+    // If a process with the designated ID is not found, the returned Process object should be a stub, and isAlive should return false.
+    @Test
+    public void noSuchProcessTest() {
+        Process process = new Process(-1);
+        Assertions.assertFalse(process.isAlive());
+    }
+
+    // After starting a sleeping process in a new thread, isAlive should return true. After killing the process, it should return false.
+    @Test
+    public void isAliveTest() {
+        Assertions.assertDoesNotThrow(() -> {
+            ProcessBuilder pb = new ProcessBuilder();
+            pb.command("sleep", "10");
+
+            TestThread processThread = new TestThread(pb);
+            processThread.run();
+
+            ArrayList<Integer> pids = getPidByCommand("sleep");
+
+            Process sleepingProcess = new Process(pids.get(0));
+
+            Assertions.assertTrue(sleepingProcess.isAlive());
+
+            java.lang.Process kill = Runtime.getRuntime().exec("kill " + pids.get(0));
+            kill.waitFor();
+
+            Thread.sleep(10);
+            Assertions.assertFalse(sleepingProcess.isAlive());
+
+        });
+    }
+
+    // The number of tasks should be equal to the number of threads in /proc/pid/stat. Test for both systemd and the current JVM
```

Comment:
Tests two things -> should be two different tests as the content of pid 1 and jvm stats do not depend on each other